### PR TITLE
Problem list linker table

### PR DESCRIPTION
**Pull request by t-young31**
## Reason for changes and summary

Adds a processing suitable for a PatientCondition <-> Hospital visit linker table. See [Emap-Core PR](https://github.com/inform-health-informatics/Emap-Core/pull/210) for details.

## Checklist for submitter & reviewer

- [ ] copy constructor copies the primary key Id of the entity
- [ ] copy constructor copies all other fields of the entity, starting with `super(other)`
- [ ] unique constraints added if appropriate
- [ ] text definition added to columns which could have more than 255 characters
- [x] foreign key names are in the form `<entityClass>Id`, and no typos
- [ ] during run, glowroot is checked for query speed and indexes added if needed
- [ ] validation run completed and validated

### CODE REVIEW

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/PatientConditionVisitLink.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * between the two.
+ *
+ * @author Tom Young
+ */
+@Entity
+@Data
+@EqualsAndHashCode
+@NoArgsConstructor
+@ToString(callSuper = true)
+public class PatientConditionVisitLink {
```
Comment (stefpiatek) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/PatientConditionVisitLink.java:28]: could maybe be ConditionVisits? or PatientConditionVisits? Though it usually takes me a few iterations to get a name that I'm happy with

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/PatientConditionVisitLink.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    private Long patientConditionVisitLinkId;
+
+    /**
+     * \brief Identifier for the HospitalVisit associated with this record.
+     *
+     * This is a foreign key that joins to the HospitalVisit table.
+     */
+    @ManyToOne
+    @JoinColumn(name = "hospitalVisitId", nullable = false)
+    private HospitalVisit hospitalVisitId;
```
Comment (stefpiatek) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/PatientConditionVisitLink.java:46]: had a quick look at `star_validation` for problem lists and we do get problem lists where there isn't a hospital visit, so this should be nullable

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/PatientConditionVisitLink.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ * Linker table between the PatientCondition and HospitalVisit tables to efficiently model the many-many relationship
+ * between the two.
+ *
+ * @author Tom Young
+ */
+@Entity
+@Data
+@EqualsAndHashCode
+@NoArgsConstructor
+@ToString(callSuper = true)
```
Comment (stefpiatek) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/PatientConditionVisitLink.java:27]: probably worth adding an index to each of the FK columns

**Review by stefpiatek — APPROVED** [review]
Looks good, maybe some naming and constaint and index rejigging

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/PatientConditionVisitLink.java (hunk 2 above)
Comment (t-young31) [reply inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/PatientConditionVisitLink.java:46]: ah yeah – the processing doesn't add a link if the visit is `null`. seems silly to link to a null object?!

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/PatientConditionVisitLink.java (hunk 1 above)
Comment (t-young31) [reply inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/PatientConditionVisitLink.java:28]: the most difficult thing in programming 😉

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/PatientConditionVisitLink.java (hunk 2 above)
Comment (stefpiatek) [reply inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/PatientConditionVisitLink.java:46]: 👍 should have read that one first, ignore me!

**Pull request merged**